datafusion/common/src/functional_dependencies.rs

```diff
@@ -68,32 +68,48 @@ impl Constraints {
         let constraints = constraints
             .iter()
             .map(|c: &TableConstraint| match c {
-                TableConstraint::Unique {
-                    columns,
-                    is_primary,
-                    ..
-                } => {
+                TableConstraint::Unique { name, columns, .. } => {
                     let field_names = df_schema.field_names();
-                    // Get primary key and/or unique indices in the schema:
+                    // Get unique constraint indices in the schema:
                     let indices = columns
                         .iter()
-                        .map(|pk| {
+                        .map(|u| {
                             let idx = field_names
                                 .iter()
-                                .position(|item| *item == pk.value)
+                                .position(|item| *item == u.value)
                                 .ok_or_else(|| {
+                                    let name = name
+                                        .as_ref()
+                                        .map(|name| format!("with name '{name}' "))
+                                        .unwrap_or("".to_string());
                                     DataFusionError::Execution(
-                                        "Primary key doesn't exist".to_string(),
+                                        format!("Column for unique constraint {}not found in schema: {}", name,u.value)
```

Comment:
```suggestion
                                        format!("Column for unique constraint {} not found in schema: {}", name,u.value)
```

Reply:
Ah the missing whitespace is actually intentional here as the unique constraint may be unnamed :slightly_smiling_face:

Reply:
Got it -- sorry -- maybe we can add a test or something of the error message in a follow on PR to make it clearer.